src/main/java/de/tum/cit/aet/artemis/quiz/service/QuizSubmissionService.java

```diff
@@ -176,53 +186,41 @@ else if (quizExercise.isQuizEnded()) {
-        var user = participation.getParticipantIdentifier();
-        removeUnnecessaryObjectsBeforeSendingToClient(participation);
-        websocketMessagingService.sendMessageToUser(user, "/topic/exercise/" + quizExerciseId + "/participation", participation);
-    }
+    /**
+     * Fetches quiz submissions for the given student participations and associates them with the participations.
+     *
+     * @param participations The set of student participations to associate with quiz submissions.
+     */
+    private void associateQuizSubmissionsWithStudentParticipations(Set<StudentParticipation> participations) {
+        Set<Long> participationIds = participations.stream().map(StudentParticipation::getId).collect(Collectors.toSet());
 
-    // TODO: Use a DTO instead of removing data from the entity
-    @Deprecated
-    private void removeUnnecessaryObjectsBeforeSendingToClient(StudentParticipation participation) {
-        if (participation.getExercise() != null) {
-            var quizExercise = (QuizExercise) participation.getExercise();
-            // we do not need the course and lectures
-            quizExercise.setCourse(null);
-        }
-        // submissions are part of results, so we do not need them twice
-        participation.setSubmissions(null);
-        participation.setParticipant(null);
-        if (participation.getResults() != null && !participation.getResults().isEmpty()) {
-            QuizSubmission quizSubmission = (QuizSubmission) participation.getResults().iterator().next().getSubmission();
-            if (quizSubmission != null && quizSubmission.getSubmittedAnswers() != null) {
-                for (SubmittedAnswer submittedAnswer : quizSubmission.getSubmittedAnswers()) {
-                    if (submittedAnswer.getQuizQuestion() != null) {
-                        // we do not need all information of the questions again, they are already stored in the exercise
-                        var question = submittedAnswer.getQuizQuestion();
-                        submittedAnswer.setQuizQuestion(question.copyQuestionId());
-                    }
-                }
-            }
+        List<QuizSubmission> submissions = quizSubmissionRepository.findWithEagerSubmittedAnswersByParticipationIds(participationIds);
+
+        Map<Long, Set<QuizSubmission>> submissionsByParticipationId = submissions.stream().collect(Collectors.groupingBy(s -> s.getParticipation().getId(), Collectors.toSet()));
+
+        for (StudentParticipation participation : participations) {
+            Set<QuizSubmission> quizSubmissions = submissionsByParticipationId.getOrDefault(participation.getId(), Set.of());
+            Set<Submission> submissionSet = new HashSet<>(quizSubmissions);
+            participation.setSubmissions(submissionSet);
         }
     }
 
+    private void sendQuizResultToUser(long quizExerciseId, StudentParticipation participation) {
+        var user = participation.getParticipantIdentifier();
```

Comment:
```suggestion
        String user = participation.getParticipantIdentifier();
```

Reply:
Yes here with `var` I'd assume it was of type `User`. Specifying it here makes sense